airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSourceOperations.java

```diff
@@ -67,58 +72,87 @@ public JsonNode rowToJson(final ResultSet queryContext) throws SQLException {
+            case BIGINT -> putBigInt(json, columnName, resultSet, colIndex);
+            case FLOAT, DOUBLE -> putDouble(json, columnName, resultSet, colIndex);
+            case REAL -> putFloat(json, columnName, resultSet, colIndex);
+            case NUMERIC, DECIMAL -> putBigDecimal(json, columnName, resultSet, colIndex);
+            // BIT is a bit string in Postgres, e.g. '0100'
+            case BIT, CHAR, VARCHAR, LONGVARCHAR -> putString(json, columnName, resultSet, colIndex);
+            case DATE -> putDate(json, columnName, resultSet, colIndex);
+            case TIME -> putTime(json, columnName, resultSet, colIndex);
+            case TIMESTAMP -> putTimestamp(json, columnName, resultSet, colIndex);
+            case BLOB, BINARY, VARBINARY, LONGVARBINARY -> putBinary(json, columnName, resultSet, colIndex);
+            case ARRAY -> putArray(json, columnName, resultSet, colIndex);
+            default -> putDefault(json, columnName, resultSet, colIndex);
+          }
+        }
       }
     }
   }
 
+  @Override
+  protected void putDate(ObjectNode node, String columnName, ResultSet resultSet, int index) throws SQLException {
+    LocalDate date = getDateTimeObject(resultSet, index, LocalDate.class);
+    node.put(columnName, resolveEra(date, date.toString()));
+  }
+
+  @Override
+  protected void putTime(ObjectNode node, String columnName, ResultSet resultSet, int index) throws SQLException {
+    LocalTime time = getDateTimeObject(resultSet, index, LocalTime.class);
+    node.put(columnName, time.toString());
+  }
+
+  @Override
+  protected void putTimestamp(ObjectNode node, String columnName, ResultSet resultSet, int index) throws SQLException {
+    LocalDateTime timestamp = getDateTimeObject(resultSet, index, LocalDateTime.class);
+    LocalDate date = timestamp.toLocalDate();
+    node.put(columnName, resolveEra(date, timestamp.toString()));
+  }
+
   @Override
   public JDBCType getFieldType(final JsonNode field) {
     try {
-      final String typeName = field.get(INTERNAL_COLUMN_TYPE_NAME).asText();
+      final String typeName = field.get(INTERNAL_COLUMN_TYPE_NAME).asText().toLowerCase();
       // Postgres boolean is mapped to JDBCType.BIT, but should be BOOLEAN
-      if (typeName.equalsIgnoreCase("bool") || typeName.equalsIgnoreCase("boolean")) {
-        return JDBCType.BOOLEAN;
-      } else if (typeName.equalsIgnoreCase("bytea")) {
+      return switch (typeName) {
+        case "bool", "boolean" -> JDBCType.BOOLEAN;
         // BYTEA is variable length binary string with hex output format by default (e.g. "\x6b707a").
         // It should not be converted to base64 binary string. So it is represented as JDBC VARCHAR.
         // https://www.postgresql.org/docs/14/datatype-binary.html
-        return JDBCType.VARCHAR;
-      }
-
-      return JDBCType.valueOf(field.get(INTERNAL_COLUMN_TYPE).asInt());
+        case "bytea" -> JDBCType.VARCHAR;
+        case "timestamptz" -> JDBCType.TIMESTAMP_WITH_TIMEZONE;
```

Comment:
Strings `timestamptz` and `timetz` are used more than once in the codebase - please create `public static` constants for them.

Reply:
> Strings `timestamptz` and `timetz` are used more than once in the codebase - please create `public static` constants for them.

done